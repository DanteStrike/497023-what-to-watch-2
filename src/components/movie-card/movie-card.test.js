import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

it(`Render correctly MovieCard component`, () => {
  const videoOptionsMock = {
    poster: ``,
    isPlaying: false,
    isMuted: true,
    src: ``
  };

  const MovieCardComponent = renderer
    .create(
        <MovieCard
          id={2}
          title={`Johnny English`}
          image={`img/johnny-english.jpg`}
          titleLinkHref={`#`}
          onFilmMouseHover={jest.fn()}
          onFilmMouseLeave={jest.fn()}

          videoPlayerOptions={videoOptionsMock}
        />,
        {
          createNodeMock: (element) => {
            if (element.type === `video`) {
              return {
                src: null,
                isMuted: null,
                poster: null
              };
            }
            return null;
          }
        }
    ).toJSON();

  expect(MovieCardComponent).toMatchSnapshot();
});
