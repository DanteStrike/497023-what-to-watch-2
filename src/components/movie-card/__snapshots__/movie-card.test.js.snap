// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly MovieCard component 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      style={
        Object {
          "display": "block",
          "height": "100%",
          "objectFit": "fill",
          "width": "100%",
        }
      }
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      Johnny English
    </a>
  </h3>
</article>
`;
